docs/DesignDocs/AttestedTLSAPIs.md

```diff
@@ -0,0 +1,758 @@
+the enclave evidence embedded, which then is used during
+the TLS handshake. More importantly, the API binds the
+public key of the certificate with the evidence such that
+the TLS channel based on the public key can be *attested*.
+To validate the certificate, the receiver uses the
+`oe_verify_attestation_certificate_with_evidence_v2`
+API, which assesses the trustworthiness of the evidence
+(based on the DCAP client). In addition, the API outputs a
+list of claims, allowing the developer to implement additional
+verification logic (e.g., authenticate the enclave identity, or `MRENCLAVE`).
+After a successful validation, the receiver continues the handshake
+and finally establishes an *attested* secure channel with the sender.
+
+While the two APIs are useful in the above end-to-end scenario,
+they have the following limitations:
+
+- Fail to support general use cases
+
+  A typical attestation procedure includes three roles: an attester, a relying party,
+  and a verifier. Two general models are as follows.
+  - Passport Model
+
+    The attester generates the evidence and asks the verifier to assess the
+    trustworthiness of the evidence. Then, the attester sends the attestation results
+    to the relying party. Finally, the relying party validates the attestation results
+    based on customized policies.
+
+  - Background-Check Model
+
+    The attester generates the evidence and directly sends the evidence to the
+    relying party. The relying party then asks the verifier to assess the trustworthiness
+    of the evidence and then validates the attestation results based on customized policies.
+
+  The APIs, however, only support the special case of the background-check model where
+  the sender of the certificate is the attester and the receiver of the certificate plays
+  the roles of both the verifier and the relying party. More specifically, the
+  `oe_get_attestation_certificate_with_evidence_v2` API internally obtains
+  the raw enclave evidence and embeds it into the output certificate while the
+  `oe_verify_attestation_certificate_with_evidence_v2` API takes the certificate
+  as input, extracts and verifies the evidence internally based on the local DCAP
+  client, and outputs the list of the claims. As a result, neither can the sender
+  embed the attestation results into the certificate nor can the receiver extract
+  the raw evidence from the certificate and ask the verifier to assess the evidence,
+  thus failing to support the two abovementioned models.
+
+- Lack of support of init-time custom claims
+
+  According to [Open Enclave Init-time Configuration Interface](InitTimeConfigurationInterface.md),
+  the support of init-time custom claims is desirable when the TEE hardware has such
+  support (e.g., `CONFIGID` in SGX2). The `oe_get_attestation_certificate_with_evidence_v2`
+  API does not allow developers to provide an init-time custom claim buffer as input.
```

Comment:
The [docs](https://github.com/ietf-rats-wg/architecture/blob/master/draft-ietf-rats-architecture.md) only mentions the term "claims".

Reply:
Right because I think the notion of "init time custom claims" is really an SGX-specific artifact that RATS solves in a more general way, if I understand init time custom claims correctly.  In RATS layered attestation, configuration data can be another layer from the enclave itself.  SGX and OE historically has not supported layered attestation and instead had (has) an SGX-only model.  Once you support layered attestation, you have a much more general solution that does not need any special term for this.